lib/matplotlib/backends/backend_gtk3.py

```diff
@@ -827,6 +848,8 @@ class _BackendGTK3(_Backend):
 
     @staticmethod
     def mainloop():
-        if Gtk.main_level() == 0:
-            cbook._setup_new_guiapp()
-            Gtk.main()
+        global _application
+
+        _application.run()  # Quits when all added windows close.
```

Comment:
I think `show` only calls `mainloop` if there's at least an active figuremanager, and thus `_application` should indeed never be None, but perhaps you can call _create_application() here first to be on the safe side?

Reply:
I modelled this after Qt, which doesn't seem to do that. Though I guess it creates the `QApplication` in `FigureCanvasQt`, so maybe that's why it's not necessary there?

I did assume, as you said, that `mainloop` would only be called if there were managers.

Reply:
Creating the GtkApplication in `FigureCanvasGTK3.__init__` (as for qt) does seem more robust, I guess.

Reply:
I agree we should gracefully handle the `None` case.  I think the best we can do is fast return.